### DISCUSSION TITLE
Epics Control Configuration Management

### DISCUSSION THREAD

**Discussion by amichelotti** [epics-containers/epics-containers.github.io#72] [Ideas]
Hi there,
being this transition from our custom control system to Epics, a hot feature that is not clear the epics world has is a centralize way to collect informations about the
IOC (parameters, initialization values,modules, infos), provide information where is running (in case of containers should provide info about the container/helm parameters..) the devices controlled by the IOC, and the relationships between the IOC and devices, and so on. A sort of ' control asset management'. So crucial as soon the number of devices and iocs grows, and to quickly find the possibile source of problems in case of faults or malfunctions. It seemed to me that ibek + yaml/json, could be a good starting point. Do you have any idea or solutions for this at Diamond?

**Comment by gilesknap** [comment]
@coretl I believe that Ophyd will help us in this respect at the device level. I'm less clear how this collects up to a description of a beamline or whole facility. Can you comment on this please?

**Reply by coretl** [reply]
Our plan was to make things as much as possible "one IOC per device". This means that the Kubernetes dashboard (or `ec ps`) should list the devices and also the IOCs as there is a one to one mapping. This falls down for things like the vacuum system, but even then it should be obvious which IOC hosts a device. From the point of view of "my device failed, where is its IOC?", then we plan to use ChannelFinder to do the reverse lookup of an IOC from a PV. We do plan on using the ibek yaml to generate the beamline synoptic, so all motors in a beamline exist on the beamline GUI somewhere, but that is not specified in detail yet.
